File: moderate/reverse_add.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    chomp $line;
    my ( $i, $r );
    for ( $i = 0 ; $i < 100 ; $i++ ) {
        $r = reverse($line);
        last if ( $r == $line );
        $line += $r;
    }
    printf( "%s\n", $r == $line ? $i . ' ' . $line : "not found" );
}
close(INFILE);

File: moderate/reverse_add.rb
File.open(ARGV[0]).each_line do |line|
  a = line.to_i
  i = 0
  while i < 100
    r = a.to_s.reverse.to_i
    break if a == r
    a += r
    i += 1
  end
  puts a == r ? i.to_s + ' ' + a.to_s : 'not found'
end
